app/models/task.rb

```diff
@@ -7,4 +7,12 @@ class Task < ApplicationRecord
                      minimum: Settings.validates.model.task.name.min_length,
                      maximum: Settings.validates.model.task.name.max_length
                    }
+
+  scope :by_id, ->(ids){where id: ids if ids.present?}
+
+  def status_by_user_course_subject user_id, course_id
+    user_task = user_tasks.by_task_id(id).task_status(user_id, course_id)
```

Comment:
```suggestion
    user_task = user_tasks.task_status(user_id, course_id)
```
đoạn này k cần by_task nữa